src/pbft/pbft.h

```diff
@@ -181,13 +185,36 @@ namespace pbft
         std::make_unique<ClientProxy<kv::TxHistory::RequestID, void>>(
           *message_receiver_base);
 
-      auto cb = [](Reply* m, void* ctx) {
+      auto reply_handler_cb = [](Reply* m, void* ctx) {
         auto cp =
           static_cast<ClientProxy<kv::TxHistory::RequestID, void>*>(ctx);
         cp->recv_reply(m);
       };
+      message_receiver_base->register_reply_handler(
+        reply_handler_cb, client_proxy.get());
 
-      message_receiver_base->register_reply_handler(cb, client_proxy.get());
+      auto batch_committed_cb =
+        [](uint64_t batch_id, void* batch, uint32_t size, void* ctx) {
+          auto ledger = static_cast<raft::LedgerEnclave*>(ctx);
+
+          struct LedgerEntry
+          {
+            uint64_t batch_id;
+            uint8_t data[0];
+          };
+
+          // TODO: This needs to be encrypted
+          std::vector<uint8_t> entry;
+          entry.resize(sizeof(LedgerEntry) + size);
+          auto ledger_entry = reinterpret_cast<LedgerEntry*>(entry.data());
```

Comment:
Could we use `serialized::write()` instead here?

Reply:
this has been moved to the epbft code that does not have the helper functions